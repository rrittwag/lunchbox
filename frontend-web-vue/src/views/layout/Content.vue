<template>
  <main class="w-full h-full">
    <RouterView v-slot="{ Component }">
      <transition
        mode="out-in"
        leave-active-class="transition-opacity duration-150"
        leave-to-class="opacity-0"
        enter-active-class="transition-opacity duration-150"
        enter-from-class="opacity-0"
      >
        <component :is="Component" />
      </transition>
    </RouterView>
  </main>
</template>
