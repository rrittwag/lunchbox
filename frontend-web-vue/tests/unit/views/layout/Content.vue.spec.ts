import Content from '@/views/layout/Content.vue'
import { mount } from '@vue/test-utils'
import { createRouterMock, injectRouterMock } from 'vue-router-mock'
import { RouterView } from 'vue-router'

describe('Content', () => {
  const router = createRouterMock({})
  beforeEach(() => {
    injectRouterMock(router)
  })

  test('renders snapshot', () => {
    const wrapper = mount(Content)

    expect(wrapper.element).toMatchSnapshot()
  })

  test('contains main and RouterView', () => {
    const wrapper = mount(Content)

    const mainTag = wrapper.get('main')
    mainTag.getComponent(RouterView) // throws error if not existing
  })
})
