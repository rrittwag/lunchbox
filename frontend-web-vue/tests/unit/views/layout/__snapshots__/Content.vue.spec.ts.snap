// Vitest Snapshot v1

exports[`Content > renders snapshot 1`] = `
<main
  class="h-full w-full"
>
  <router-view-stub
    name="default"
  />
</main>
`;

exports[`Content renders snapshot 1`] = `
<main
  class="h-full w-full"
>
  <router-view-stub
    name="default"
  />
</main>
`;
